### PR TITLE
refactor: Extensibility improvement

### PR DESCRIPTION
**Pull request by HarisChoudhary**

### CODE REVIEW

File: pkg/threeport-installer/v0/components.go [hunk 1]
```diff
@@ -56,7 +57,11 @@ func (cpi *ControlPlaneInstaller) UpdateThreeportAPIDeployment(
 ) error {
 	apiImage := cpi.getImage(cpi.Opts.RestApiInfo.Name, cpi.Opts.RestApiInfo.ImageName, cpi.Opts.RestApiInfo.ImageRepo, cpi.Opts.RestApiInfo.ImageTag)
 	apiArgs := cpi.getAPIArgs()
-	apiVols, apiVolMounts := cpi.getAPIVolumes()
+	apiVols, apiVolMounts, err := cpi.getAPIVolumes()
+	if err != nil {
+		return fmt.Errorf("could not get vols: %w", err)
```
Comment (HarisChoudhary) [inline pkg/threeport-installer/v0/components.go:62]: Make it specific to api vol

File: pkg/threeport-installer/v0/installer.go [hunk 2]
```diff
@@ -9,39 +9,38 @@ type InstallerOption func(o *Options)
 type CustomInstallFunction func(*v0.KubernetesRuntimeInstance, *ControlPlaneInstaller) error
 
 type Options struct {
```
Comment (HarisChoudhary) [inline pkg/threeport-installer/v0/installer.go:11]: Add comments for the each field

File: pkg/threeport-installer/v0/installer.go [hunk 3]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	// Info for the Rest Api being installed
+	RestApiInfo *v0.ControlPlaneComponent
+
+	// Info for the agent being installed
+	AgentInfo *v0.ControlPlaneComponent
+
+	// A boolean used to indicate whether the installer is being run from within threeport itself such as a reconciler
+	InThreeport bool
+
+	// CreateOrUpdate Kube resources during install
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:36]:
```suggestion
	// CreateOrUpdate Kube resources during install. If true, resources will be updated if they already exist. If false, an error will occur if a resource already exists.
```

File: pkg/threeport-installer/v0/installer.go [hunk 4]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	// Installer option to determine auth is enabled/disabled
+	AuthEnabled bool
+
+	// The AWS config profile to draw credentials from when using eks provider.
+	AwsConfigProfile string
+
+	// Retrieve AWS credentials from environment variables when using eks provider.
+	AwsConfigEnv bool
+
+	// AWS region code to install threeport runtimes in.
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:48]:
```suggestion
	// AWS region code to install threeport control plane in.
```

File: pkg/threeport-installer/v0/installer.go [hunk 5]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	// Email address of control plane admin. Provided to TLS provider.
+	CreateAdminEmail string
+
+	// Bool used to indicate whether installing in Dev environment or not
+	DevEnvironment bool
+
+	// EncryptionKey is the key used to encrypt and decrypt sensitive fields.
+	EncryptionKey string
+
+	// Overwrite any applicable config entried
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:66]:
```suggestion
	// Overwrite any applicable config entries
```

File: pkg/threeport-installer/v0/installer.go [hunk 6]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	// InfraProvider to instal control plane on e.g. kind, eks etc
+	InfraProvider string
+
+	// Path to kube config
+	KubeconfigPath string
+
+	// Number of additional worker nodes to deploy. Only applies to kind provider. (default is 0)
+	NumWorkerNodes int
+
+	// Path to infra provider config directory
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:81]:
```suggestion
	// Path to infra provider config directory where cloud infra inventory is saved.
```

File: pkg/threeport-installer/v0/installer.go [hunk 7]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	ProviderConfigDir string
+
+	// Path to threeport repository root
+	ThreeportPath string
+
+	// Indicate debug mode
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:87]:
```suggestion
	// If true, run in debug mode. Appropriate for development environments only.
```

File: pkg/threeport-installer/v0/installer.go [hunk 8]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	Debug bool
+
+	// Indicate live reload
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:90]:
```suggestion
	// If true, live changes made in development will be live-reloaded into control plane components.
```

File: pkg/threeport-installer/v0/installer.go [hunk 9]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	LiveReload bool
+
+	// Indicate whether only control plane components should be worked on
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:93]:
```suggestion
	// If true, infrastructure is not provisioned, control plane is installed on existing infra.
```

File: pkg/threeport-installer/v0/installer.go [hunk 10]
```diff
@@ -9,39 +9,101 @@ type InstallerOption func(o *Options)
...
+	ControlPlaneOnly bool
+
+	// Port forwards for kind infra provider
+	KindInfraPortForward []string
+
+	// Deployed EKS load balancer
```
Comment (lander2k2) [inline pkg/threeport-installer/v0/installer.go:99]:
```suggestion
	// If true, an EKS load balancer is provisioned for the threeport API.
```

**Review by lander2k2 — APPROVED** [review]

File: pkg/threeport-installer/v0/installer.go (hunk 8 above)
Comment (randalljohnson) [reply pkg/threeport-installer/v0/installer.go:90]: I would also add that this is only relevant if the infra provider is `kind`

**Review by randalljohnson — APPROVED** [review]

**Pull request merged**